client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -135,6 +136,8 @@ function mediaButton( editor ) {
 		return { isLoaded, onLoad };
 	} )();
 
+	const mediaHasChanged = ( media ) => withoutHttp( media.guid ) !== withoutHttp( media.URL );
```

Comment:
Haha, nice. Nono, we need it. Yeah, for the line 184 (`	} else if ( ! mediaHasChanged( media ) ) {`), it doesn't do much (as you've pointed out), but for the media width/height, it's still important.

Try this:
1. Drag and drop new media of bigger size (>2mb) or throttle your network connection;
2. Resize the transient media which was inserted into the post editor;
3. Unthrottle your network and watch the media get uploaded;
4. Its size in post editor shouldn't change;
5. Now, edit the image in image editor to a very small size (smaller than the size of the inserted image in post editor);
6. The size of the inserted image should change to reflect the size change when you edited the image;
7. On the other hand, if the edited image has larger size than the inserted image, don't change the size of the inserted image.